draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -537,447 +539,170 @@ Note: Examples in this document use `cnf` Claim defined in [@RFC7800] to include
-         digests match.
-      6. Ensure that the claim value in the HS-Disclosures JWT is a JSON-encoded
-         object containing at least the keys `s` and `v`, and optionally `n`.
-      7. Store the value of the key `v` as the claim value. If `n` is contained
-         in the object, use the value of the key `n` as the claim name.
-    3. Once all necessary claims have been verified, their values can be
-       validated and used according to the requirements of the application. It
-       MUST be ensured that all claims required for the application have been
-       disclosed.
-
-If any step fails, the input is not valid and processing MUST be aborted.
-
-## Processing Model {#processing_model}
-
-Neither an SD-JWT nor an HS-Disclosures JWT is suitable for direct use by an application.
-Besides the REQUIRED verification steps listed above, it is further RECOMMENDED
-that an application-consumable format is generated from the data released in
-the HS-Disclosures. The RECOMMENDED way is to merge the released claims and any
-plaintext claims in the SD-JWT recursively:
-
- * Objects from the released claims must be merged into existing objects from the SD-JWT.
- * If a key is present in both objects:
-   * If the value in the released claims is an object and the value in the
-     SD-JWT claims is an object, the two objects MUST be merged recursively.
-   * Else, the value in the released claims MUST be used.
-
-The keys `sd_digests` and `sd_digest_derivation_alg` SHOULD be removed prior to further
-processing.
-
-The processing is shown in Examples 2b and 3 in the Appendix.
+    5. Check that the `sd_digest_derivation_alg` claim is present and its value is understood and the digest derivation algorithm is deemed secure.
+ 4. Create a copy of the SD-JWT payload for further processing, if required.
+ 5. Process the Disclosures. For each Disclosure provided:
+    1. Calculate the hash digest over the base64url string as described in (#hashing_disclosures).
+    2. Find any `_sd` keys in the SD-JWT payload where the digest of the Disclosure is included in the array.
+       1. If the digest cannot be found in the SD-JWT payload, the Verifier MUST reject the Presentation.
+       2. If there is more than one place where the digest is included, the Verifier MUST reject the Presentation.
+       3. If there is a key `_sd` that does not refer to an array, the Verifier MUST reject the Presentation.
+       4. Otherwise, insert, at the level of the `_sd` claim, the claim described by the Disclosure with the claim name and claim value provided in the Disclosure.
+          1. If the Disclosure is not a JSON-encoded array of three elements, the Verifier MUST reject the Presentation.
+          2. If the claim name already exists, the Verifier MUST reject the Presentation. Note that this also means that if a Holder sends the same Disclosure multiple times, the Verifier MUST reject the Presentation.
```

Comment:
It can, but it must not already exist at the same level. Note that in this version of the draft there's no merging defined (as it is not strictly needed).

Reply:
```suggestion
          2. If the claim name already exists at the same level, the Verifier MUST reject the Presentation. Note that this also means that if a Holder sends the same Disclosure multiple times, the Verifier MUST reject the Presentation.
```